rumqttd/src/lib.rs

```diff
@@ -181,13 +187,22 @@ impl Broker {
             })?;
         }
 
+        let mut runtime = tokio::runtime::Builder::new_current_thread();
+        let runtime = runtime.enable_all().build().unwrap();
+
         // Run console in current thread, if it is configured.
         if self.config.console.is_some() {
             let console = ConsoleLink::new(self.config.clone(), self.router_tx.clone());
             let console = Arc::new(console);
-            consolelink::start(console);
+            runtime.spawn(async {
+                consolelink::start(console).await;
+            });
         }
 
+        runtime.block_on(async {
+            signal::ctrl_c().await.unwrap();
+        });
```

Comment:
Waiting for CTRL+C really shouldn't be part of the server - especially in a non-configurable way. At the very least, it should be toggle-able, but I would much rather the responsibility of waiting and blocking was on the embedder. As of now, it interferes with a SIGINT handler that the embedder sets up and silently swallows it.

Reply:
Yeah I agree that this is not great. But I need to find out if it's possible to output profiling information after ctrl + c